### DISCUSSION TITLE
Reconcile on each commit to manifest repository

### DISCUSSION THREAD

**Discussion by joeshadbolt** [fluxcd/flux2#3482] [Q&A] [answered]
We are using a [repo per environment](https://fluxcd.io/flux/guides/repository-structure/#repo-per-environment) structure for our manifests, which contains a number of HelmRelease manifests on a per applicatio basis typically. 

We'd like flux to ultimately perform a reconciliation per commit to the manifests repository, so we can feedback to an Azure DevOps pipeline which is associated with that commit.

Currently the source controller is configured to poll the repository on an interval and it appears the reconciliation metadata for the git commit only represents HEAD, therefore losing the metadata for any intermediary commits before HEAD and since the source controller last synced.

![image](https://user-images.githubusercontent.com/4125403/212076137-5761e9bb-99d2-488b-ab20-366d03f8d678.png)

Ignoring some obvious questions why exactly we may want to do this, is it possible to treat each commit to a monitored branch individually and sequentially with the source controller?

**Comment by makkes** [comment, marked as answer]
Reconciliation on a per-commit basis is not possible with source-controller out of the box because of its polling nature.

What you could possibly try is to use an [Incoming Webhook Receiver](https://fluxcd.io/flux/components/notification/receiver/) that would trigger a reconciliation upon each commit. However, that obviously doesn't guarantee that the reconciliation will happen in order or in sequence, depending on the frequency of new commits.

**Reply by joeshadbolt** [reply]
Thanks for your response and suggestion.
